core/metadata.h

```diff
@@ -119,6 +119,11 @@ class Pipeline {
   // Main entry point for calculating metadata offsets.
   int ComputeMetadataOffsets();
 
+  // Registers attr and returns 0 if no attribute named attr->name with size
+  // other than attr->size has already been registered for this pipeline.
+  // Returns -errno on error.
+  int RegisterAttribute(const struct mt_attr *attr);
+
```

Comment:
DeRegisterAttribute() would be nice to have, but I'm not sure there's a clean way to do that without allowing any module to delete previously registered attributes